preinstall.js

```diff
@@ -0,0 +1,75 @@
+const fs = require('fs')
+const path = require('path')
+var request = require('request')
+const { libcoreVersion } = require('./package.json')
+
+const perPlatform = {
+  linux: {
+    dir: 'linux',
+    files: ['libledger-core.so'],
+    chmod: 0o755,
+  },
+  darwin: {
+    dir: 'macos',
+    files: ['libledger-core.dylib'],
+    chmod: 0o755,
+  },
+  win32: {
+    dir: 'win/vs2015',
+    file: ['ledger-core.dll', 'ledger-core.lib', 'crypto.dll'],
+  },
+}
+
+const conf = perPlatform[process.platform]
+if (!conf) {
+  console.error(`Platform ${process.platform} is not supported`)
+  process.exit(1)
+}
+const endpointURL = `https://s3-eu-west-1.amazonaws.com/ledger-lib-ledger-core/${libcoreVersion}/${
+  conf.dir
+}`
+
+if (!fs.existsSync('lib')) {
+  fs.mkdirSync('lib')
+}
+
+conf.files.reduce(
```

Comment:
was it intentionally sequential versus `Promise.all(conf.files.map(file => ...))`?

Reply:
More for the logs to be sequential, also consistent with previous code. Not really important because only windows have more than one file and it's just 3 files